data_analysis/data_stats.py

```diff
@@ -22,15 +22,19 @@ class Stats:
     # Graph?
     stats = {'all': {}, 'new': {}}
 
-    def __init__(self, max_date=datetime.now() - timedelta(hours=3)):
+    def __init__(self):
         from helpers import RtdRay
 
-        min_date = max_date + timedelta(1)
         self._rtd_d = RtdRay().load_data(columns = ["dp_delay", "ar_delay", "dp_pt", "ar_pt", "ar_cs", "dp_cs"])
+        
+        max_date = self._rtd_d['ar_pt'].max().compute()
+        min_date = max_date - timedelta(1)
+        print("Last 24h from %s to %s" % (min_date, max_date))
+
         # HYPER OPTIMISATION SPEED DATA LOADING
         self._rtd_d_new = self._rtd_d.loc[
-            (self._rtd_d["ar_pt"] >= min_date) | (self._rtd_d["dp_pt"] >= min_date) |
-            (self._rtd_d["ar_pt"] <= min_date) | (self._rtd_d["dp_pt"] <= min_date)
+            ((self._rtd_d['ar_pt'] >= min_date) | (self._rtd_d['dp_pt'] >= min_date)) 
```

Comment:
RtdRay().load_data() hat ein min und max date argument, was genau das hier macht. Das kannst du also vereinfachen

Reply:
ich update da mal des in rtdray, sodass man da wenn `ar_pt` und `dp_pt` geladen sind nicht nochmal extra die daten lädt ok?

Reply:
Ich verstehe jetzt nicht was du da meinst muss ich ehrlich sagen

Reply:
```py
if 'ar_pt' in rtd.columns and 'dp_pt' in rtd.columns:
                _filter = rtd
            else:
                _filter = dd.read_parquet(self.DATA_CACHE_PATH, engine='pyarrow', columns=['ar_pt', 'dp_pt'])
```

Reply:
des sollte doch gehen oder?

Reply:
jo, mach aber `_filter = rtd.loc[:, ['ar_pt', 'dp_pt']]`